internal/buildpackage/builder.go

```diff
@@ -112,25 +112,27 @@ func (p *PackageBuilder) Save(repoName string, publish bool) (imgutil.Image, err
 }
 
 func validateBuildpacks(mainBP dist.Buildpack, depBPs []dist.Buildpack) error {
-	depsWithRefs := map[dist.BuildpackInfo][]dist.BuildpackInfo{}
+	depsWithRefs := map[string][]dist.BuildpackInfo{}
 
 	for _, bp := range depBPs {
-		depsWithRefs[bp.Descriptor().Info] = nil
+		depsWithRefs[bp.Descriptor().Info.FullName()] = nil
 	}
 
-	for _, bp := range append([]dist.Buildpack{mainBP}, depBPs...) {
+	for _, bp := range append([]dist.Buildpack{mainBP}, depBPs...) { // List of everything
 		bpd := bp.Descriptor()
 		for _, orderEntry := range bpd.Order {
 			for _, groupEntry := range orderEntry.Group {
-				if _, ok := depsWithRefs[groupEntry.BuildpackInfo]; !ok {
+				if _, ok := depsWithRefs[groupEntry.BuildpackInfo.FullName()]; !ok {
 					return errors.Errorf(
-						"buildpack %s references buildpack %s which is not present",
+						"buildpack %s references buildpack %s which is not present\nmap: %+v\nlooking-for: %+v\n",
```

Comment:
This additional information seems excessive and more like debugging details.

Reply:
hah it was...